File: app/src/main/java/com/oralvis/oralvisclient/domain/model/Booking.kt
package com.oralvis.oralvisclient.domain.model

/**
 * Appointment/booking (from appointments list, bookings-by-date, etc.).
 */
data class Booking(
    val id: String,
    val patientName: String,
    val appointmentDate: String,
    val slotTime: String,
    val status: BookingStatus,
    val notes: String? = null,
    val paymentId: String? = null,
    val durationMinutes: Int? = null,
    val clinicId: String? = null,
    val patientId: String? = null,
    val walkinPatientId: String? = null,
    val amountPaid: Double? = null,
    val refundStatus: String? = null
)

enum class BookingStatus {
    PENDING,
    CONFIRMED,
    COMPLETED,
    CANCELLED,
    PAID,
    REFUND_REQUESTED,
    REFUNDED,
    CANCELLED_NO_REFUND,
    UNKNOWN
}

fun String.toBookingStatus(): BookingStatus = when (this.lowercase()) {
    "pending" -> BookingStatus.PENDING
    "confirmed" -> BookingStatus.CONFIRMED
    "completed" -> BookingStatus.COMPLETED
    "cancelled" -> BookingStatus.CANCELLED
    "paid" -> BookingStatus.PAID
    "refund-requested" -> BookingStatus.REFUND_REQUESTED
    "refunded" -> BookingStatus.REFUNDED
    "cancelled-no-refund" -> BookingStatus.CANCELLED_NO_REFUND
    else -> BookingStatus.UNKNOWN
}

File: oralvis_back/server/models/booking.js
const mongoose = require("mongoose");

const bookingSchema = new mongoose.Schema(
  {
    patient: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: false,
    },
    walkinPatient: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "WalkinPatient",
      default: null,
    },
    clinic: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Clinic",
      required: true,
    },
    appointmentDate: {
      type: Date,
      required: true,
    },
    slotTime: {
      type: String,
      required: true,
    },
    status: {
      type: String,
      enum: [
        "pending",
        "confirmed",
        "completed",
        "cancelled",
        "paid",
        "refund-requested",
        "refunded",
        "cancelled-no-refund",
      ],
      default: "pending",
    },
    notes: {
      type: String,
      default: "",
    },
    paymentId: {
      type: String,
      required: true,
    },
    refundId: {
      type: String,
      default: null,
    },
    refundStatus: {
      type: String,
      enum: ["none", "requested", "approved", "rejected"],
      default: "none",
    },
    refundRequestedAt: Date,
    refundApprovedAt: Date,
    refundRejectedAt: Date,
    amountPaid: {
      type: Number,
    },
    durationMinutes: {
      type: Number,
      default: 30,
    },
  },
  {
    timestamps: true,
  }
);

// Add index to speed up queries
bookingSchema.index({ patient: 1, clinic: 1, appointmentDate: 1, slotTime: 1 });
const Booking = mongoose.model("Booking", bookingSchema);

module.exports = Booking;
